flink-table/flink-table-common/src/test/java/org/apache/flink/table/sources/decorator/CachedLookupFunctionDecoratorTest.java

```diff
@@ -0,0 +1,248 @@
+		Assert.assertEquals(expected, multKey);
+		cachedLookupTableSource.close();
+
+	}
+
+	@Test
+	public void testWithMaxSize() throws Exception {
+		result.clear();
+		//set maxSize is 1, that means only can hold one key in cache.
+		CachedLookupFunctionDecorator<Row> cachedLookupTableSource = new CachedLookupFunctionDecorator<>(new TestLookupFucntion(),
+			1);
+		cachedLookupTableSource.setCollector(testCollector);
+		cachedLookupTableSource.open(mock(FunctionContext.class));
+
+		Cache<Row, List<Row>> cache = cachedLookupTableSource.getCache();
+		//cache still have no data.
+		Assert.assertEquals(0, cache.size());
+		cachedLookupTableSource.eval("1");
+		//load into cache and emit correctly.
+		Assert.assertEquals(1, cache.size());
+		Assert.assertEquals(5, result.size());
+		Assert.assertEquals(cache.getIfPresent(Row.of("1")), result);
+		List<Row> expected = Lists.newArrayList(Row.of("1", "0"),
+			Row.of("1", "1"),
+			Row.of("1", "2"),
+			Row.of("1", "3"),
+			Row.of("1", "4"));
+		Assert.assertEquals(expected, result);
+
+		// cache hit.
+		cachedLookupTableSource.eval("1");
+		Assert.assertEquals(1, cache.size());
+		Assert.assertEquals(10, result.size());
+		List<Row> expected2 = cache.getIfPresent(Row.of("1"));
+		expected2.addAll(cache.getIfPresent(Row.of("1")));
+		Assert.assertEquals(expected2, result);
+
+		cachedLookupTableSource.eval("2");
+		Assert.assertEquals(1, cache.size());
+		Assert.assertEquals(15, result.size());
+		expected2.addAll(cache.getIfPresent(Row.of("2")));
+		Assert.assertEquals(expected2, result);
+	}
+
+	@Test
+	public void testEvalOneObjectKey() throws Exception {
+		result.clear();
+		CachedLookupFunctionDecorator<Row> cachedLookupTableSource = new CachedLookupFunctionDecorator<>(new TestLookupFucntionOneKey(),
+			maximnCacheSize);
+		cachedLookupTableSource.setCollector(testCollector);
+		cachedLookupTableSource.open(mock(FunctionContext.class));
+
+		Cache<Row, List<Row>> cache = cachedLookupTableSource.getCache();
+		//cache still have no data.
+		Assert.assertEquals(0, cache.size());
+		cachedLookupTableSource.eval("1");
+		//load into cache and emit correctly.
+		Assert.assertEquals(1, cache.size());
+		Assert.assertEquals(2, result.size());
+		Assert.assertEquals(cache.getIfPresent(Row.of("1")), result);
+		List<Row> expected = Lists.newArrayList(Row.of("1", "0"), Row.of("1", "1"));
+		Assert.assertEquals(expected, result);
+
+		cachedLookupTableSource.eval("2");
+		Assert.assertEquals(2, cache.size());
+		Assert.assertEquals(4, result.size());
+		List<Row> expectedCache = cache.getIfPresent(Row.of("1"));
+		expectedCache.addAll(cache.getIfPresent(Row.of("2")));
+		Assert.assertEquals(expectedCache, result);
+
+		List<Row> expectedFinal = Lists.newArrayList(Row.of("1", "0"),
+			Row.of("1", "1"),
+			Row.of("2", "0"),
+			Row.of("2", "1"));
+		Assert.assertEquals(expectedFinal, result);
+	}
+
+	@Test
+	public void testLongKeyCantMatch() throws Exception {
```

Comment:
I think this target to test non-row key could not be used to eval. How about rename to `testEvalIllegallKey` ?